mycpp/gc_str.cc

```diff
@@ -482,3 +488,96 @@ List<Str*>* Str::split(Str* sep) {
 
   return result;
 }
+
+// TODO: We should reduce the number of allocations and copies done here.
+static inline Str* _StrFromFormat(const char* fmt, va_list args) {
+  int n = strlen(fmt);
+  auto beg = std::cregex_iterator(fmt, fmt + n, gStrFmtRegex);
+  auto end = std::cregex_iterator();
+  mylib::BufWriter* w = Alloc<mylib::BufWriter>();
+  for (std::cregex_iterator match = beg; match != end; ++match) {
+    auto& lit = (*match)[1];
+    assert(lit.matched);
+    w->write(StrFromC(lit.str().c_str(), lit.str().size()));
```

Comment:
how about making a variable from `lit.str()` and `width_m.str()` etc.?   I think that would read more nicely